presto-spark-base/src/main/java/com/facebook/presto/spark/execution/DetachedNativeExecutionProcessFactory.java

```diff
@@ -39,6 +40,9 @@ public class DetachedNativeExecutionProcessFactory
     private final HttpClient httpClient;
     private final ScheduledExecutorService errorRetryScheduledExecutor;
     private final JsonCodec<ServerInfo> serverInfoCodec;
+    private final JsonCodec<BatchPlanValidationRequest> planValidationRequestJsonCodec;
```

Comment:
With the list of codec growing. Can you make this a Collection of Codec `Map<Class, JsonCodec<?>>` (You can use Multibinder for this) or create a separate class containing the collection.